### PR TITLE
pythonPackages.pyroma: fix tests phase

### PR DESCRIPTION
**Pull request by kamadorueda**
- The package was not building due to an infinite
  recursion error in pytest
- After looking how they the package owners test
  it I discover they use unittest module instead
  of pytest
- Rewrite so it uses unittest

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kamadorueda** [comment]
although it causes rebuilding lots of packages, I think it can only be for the better because pyroma is already broken and therefore everything that depends on it is also broken

**Comment by SuperSandro2000** [comment]
> although it causes rebuilding lots of packages, I think it can only be for the better because pyroma is already broken and therefore everything that depends on it is also broken

Builds for me locally on master for all 3 major python versions. I think due to the amount of rebuilds (2200) we should target staging.

File: pkgs/development/python-modules/pyroma/default.nix [hunk 1]
```diff
@@ -26,23 +25,18 @@ buildPythonPackage rec {
...
-  pytestFlagsArray = [ "pyroma/tests.py" ];
-
-  disabledTests = [
-    # PyPI tests require network access
-    "PyPITest"
-  ];
+  # https://github.com/regebro/pyroma/blob/3.2/Makefile#L23
+  # PyPITest requires network access
+  checkPhase = ''
+    python -m unittest -k 'not PyPITest' pyroma.tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyroma/default.nix:31]:
```suggestion
    ${python.interpreter} -m unittest -k 'not PyPITest' pyroma.tests
```

**Comment by kamadorueda** [comment]
@SuperSandro2000 done

**Pull request merged**